homeassistant/components/plex/__init__.py

```diff
@@ -54,148 +55,94 @@
-        if host_and_port in hass.data[PLEX_DOMAIN][CONFIGURING]:
-            configurator.notify_errors(
-                hass.data[PLEX_DOMAIN][CONFIGURING][host_and_port],
-                "Failed to register, please try again.",
-            )
-            return
-
-        def plex_configuration_callback(data):
-            """Handle configuration changes."""
-            config = {
-                "host_and_port": host_and_port,
-                CONF_TOKEN: data.get("token"),
-                CONF_SSL: cv.boolean(data.get("ssl")),
-                CONF_VERIFY_SSL: cv.boolean(data.get("verify_ssl")),
-            }
-            setup_plex(configurator_info=config)
-
-        hass.data[PLEX_DOMAIN][CONFIGURING][
-            host_and_port
-        ] = configurator.request_config(
-            "Plex Media Server",
-            plex_configuration_callback,
-            description="Enter the X-Plex-Token",
-            entity_picture="/static/images/logo_plex_mediaserver.png",
-            submit_caption="Confirm",
-            fields=[
-                {"id": "token", "name": "X-Plex-Token", "type": ""},
-                {"id": "ssl", "name": "Use SSL", "type": ""},
-                {"id": "verify_ssl", "name": "Verify SSL", "type": ""},
-            ],
-        )
 
-    # End of inner functions.
+async def async_setup_entry(hass, entry):
+    """Set up Plex from a config entry."""
+    server_config = entry.data[PLEX_SERVER_CONFIG]
 
-    original_config = config
+    plex_server = PlexServer(server_config)
+    try:
+        await hass.async_add_executor_job(plex_server.connect)
+    except requests.exceptions.ConnectionError as error:
+        _LOGGER.error(
+            "Plex server (%s) could not be reached: [%s]",
+            server_config[CONF_URL],
+            error,
+        )
+        return False
+    except (
+        plexapi.exceptions.BadRequest,
+        plexapi.exceptions.Unauthorized,
+        plexapi.exceptions.NotFound,
+    ) as error:
+        _LOGGER.error(
+            "Login to %s failed, verify token and SSL settings: [%s]",
+            server_config[CONF_SERVER],
+            error,
+        )
+        return False
 
-    hass.data.setdefault(PLEX_DOMAIN, {SERVERS: {}, CONFIGURING: {}})
+    _LOGGER.debug(
+        "Connected to: %s (%s)", plex_server.friendly_name, plex_server.url_in_use
+    )
+    hass.data[PLEX_DOMAIN][SERVERS][plex_server.machine_identifier] = plex_server
 
-    if hass.data[PLEX_DOMAIN][SERVERS]:
-        _LOGGER.debug("Plex server already configured")
-        return False
+    if not hass.data.get(PLEX_MEDIA_PLAYER_OPTIONS):
+        hass.data[PLEX_MEDIA_PLAYER_OPTIONS] = MEDIA_PLAYER_SCHEMA({})
 
-    plex_config = config.get(PLEX_DOMAIN, {})
-    return setup_plex(config=plex_config)
+    for platform in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, platform)
+        )
+
+    return True
```

Comment:
Add a `async_unload_entry` function in the component, where we await forward unload of the platforms.

Any other listeners can/should be unsubscribed in `async_unload_entry`.

Reply:
I'll do this in a followup PR.